hudi-common/src/main/java/org/apache/hudi/common/util/InternalSchemaCache.java

```diff
@@ -197,15 +196,14 @@ public static InternalSchema getInternalSchemaByVersionId(long versionId, String
         .findFirst().map(f -> new StoragePath(hoodieMetaPath, f)).orElse(null);
     if (candidateCommitFile != null) {
       try {
-        byte[] data;
+        HoodieCommitMetadata metadata;
         try (InputStream is = storage.open(candidateCommitFile)) {
-          data = FileIOUtils.readAsByteArray(is);
+          metadata = commitMetadataSerDe.deserialize(instantGenerator.createNewInstant(
+                  new StoragePathInfo(candidateCommitFile, -1, false, (short) 0, 0L, 0L)),
+              is, () -> true, HoodieCommitMetadata.class);
```

Comment:
Should this be `false`, i.e., non-empty file?
```suggestion
              is, () -> false, HoodieCommitMetadata.class);
```